pkg/apply/cache/resource_cache.go

```diff
@@ -5,20 +5,39 @@ package cache
 
 import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
 	"sigs.k8s.io/cli-utils/pkg/object"
 )
 
-// ResourceCache stores unstructured resource objects in memory
+// ResourceStatus wraps an unstructured resource object, combined with the
+// computed status (whether the status matches the spec).
+type ResourceStatus struct {
+	// Resource is the last known value retrieved from the cluster
+	Resource *unstructured.Unstructured
+	// Status of the resource
+	Status status.Status
+	// StatusMessage is the human readable reason for the status
+	StatusMessage string
+}
+
+// ResourceCache stores CachedResource objects
 type ResourceCache interface {
+	ResourceCacheReader
+	// Load the resources into the cache, generating the ObjMetadata from the objects.
+	Load(...ResourceStatus) error
 	// Put the resource into the cache, generating the ObjMetadata from the object.
-	Put(obj *unstructured.Unstructured) error
+	Put(obj ResourceStatus) error
```

Comment:
What is the difference between `Load`, `Put`, and `Set`? They seem very similar.

Reply:
Load is just a wrapper for Put to add multiple atomically. Mostly just used for tests to avoid a for loop.
Put wraps Set and handles generating the ObjMetadata from the Unstructured.
Set accepts a user specified ObjMetadata as the key, since we mostly have it already when adding objects to the cache.

Reply:
I've removed Put in favor of just using Load, and then renamed Set to Put to match normal map operator naming.